src/style.rs

```diff
@@ -529,4 +529,285 @@ mod tests {
             assert_eq!(input.resolve(parent), expected);
         }
     }
+
+    mod test_flex_direction {
+        use crate::style::*;
+        use rstest::rstest;
+
+        #[rstest]
+        #[case(FlexDirection::Row, true)]
+        #[case(FlexDirection::RowReverse, true)]
+        #[case(FlexDirection::Column, false)]
+        #[case(FlexDirection::ColumnReverse, false)]
+        fn flex_direction_is_row(#[case] dir: FlexDirection, #[case] expected: bool) {
+            assert_eq!(dir.is_row(), expected);
+        }
+
+        #[rstest]
+        #[case(FlexDirection::Row, false)]
+        #[case(FlexDirection::RowReverse, false)]
+        #[case(FlexDirection::Column, true)]
+        #[case(FlexDirection::ColumnReverse, true)]
+        fn flex_direction_is_column(#[case] dir: FlexDirection, #[case] expected: bool) {
+            assert_eq!(dir.is_column(), expected);
+        }
+
+        #[rstest]
+        #[case(FlexDirection::Row, false)]
+        #[case(FlexDirection::RowReverse, true)]
+        #[case(FlexDirection::Column, false)]
+        #[case(FlexDirection::ColumnReverse, true)]
+        fn flex_direction_is_reverse(#[case] dir: FlexDirection, #[case] expected: bool) {
+            assert_eq!(dir.is_reverse(), expected);
+        }
+    }
+
+    mod test_flexbox_layout {
+        use crate::style::*;
+        use rstest::rstest;
+
+        #[rstest]
+        #[case(
+            FlexDirection::Row,
+            Size { width: Dimension::Points(1.0), height: Dimension::Points(2.0) },
+            Dimension::Points(1.0)
+        )]
+        #[case(
+            FlexDirection::Column,
+            Size { width: Dimension::Points(1.0), height: Dimension::Points(2.0) },
+            Dimension::Points(2.0)
+        )]
+        fn flexbox_layout_min_main_size(
+            #[case] dir: FlexDirection,
+            #[case] min_size: Size<Dimension>,
+            #[case] expected: Dimension,
+        ) {
+            let layout = FlexboxLayout { min_size, ..Default::default() };
+            assert_eq!(layout.min_main_size(dir), expected);
+        }
+
+        #[rstest]
+        #[case(
+            FlexDirection::Row,
+            Size { width: Dimension::Points(1.0), height: Dimension::Points(2.0) },
+            Dimension::Points(1.0)
+        )]
+        #[case(
+            FlexDirection::Column,
+            Size { width: Dimension::Points(1.0), height: Dimension::Points(2.0) },
+            Dimension::Points(2.0)
+        )]
+        fn flexbox_layout_max_main_size(
+            #[case] dir: FlexDirection,
+            #[case] max_size: Size<Dimension>,
+            #[case] expected: Dimension,
+        ) {
+            let layout = FlexboxLayout { max_size, ..Default::default() };
+            assert_eq!(layout.max_main_size(dir), expected);
+        }
+
+        #[rstest]
+        #[case(
+            FlexDirection::Row,
+            Rect { start: Dimension::Points(1.0), top: Dimension::Points(2.0), ..Default::default() },
```

Comment:
Can we use methods in these invocations? Because if so, we should consider creating some convenience methods to make using and testing these structs clearer.

Reply:
We can! I'll create some.